backend/composer/services/state_services.py

```diff
@@ -121,9 +121,16 @@ def compile_journey(connectivity_statement) -> List[str]:
            A string with each journey path description on a new line.
        """
         # Extract origins, vias, and destinations from the connectivity statement
+        Via = apps.get_model('composer', 'Via')
```

Comment:
@afonsobspinto imo we don't need this, we can use the queryset `connectivity_statement.via_set` 
and then add the prefetch related and/or select_related (if needed)

however my biggest concern here is not the prefetch/select but the many queries we do in all the loops
suggestion: query all needed data and store this in a Python "graph" (list with dicts or dict with lists and use this object to compile the path

Reply:
With use of prefetch_related there are no database queries done in the generate_paths method. I'm asserting this [here](https://github.com/MetaCell/sckan-composer/blob/feature/SCKAN-196/backend/tests/test_journey.py#L48)